src/App.jsx

```diff
@@ -14,45 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const [isAlphabeticaly, setIsAlphabeticaly] = useState(true);
+  const [isLength, setIsLength] = useState(true);
+  const [isReverse, setReverse] = useState(true);
+
+  const sortGoods = () => {
+    let newGoods = [...goodsFromServer];
+
+    if (!isAlphabeticaly) {
+      newGoods = newGoods.sort((good1, good2) => good1.localeCompare(good2));
+    }
+
+    if (!isLength) {
+      newGoods = newGoods.sort((good1, good2) => good1.length - good2.length);
+    }
+
+    if (!isReverse) {
+      newGoods = newGoods.reverse();
+    }
+
+    return newGoods;
+  };
+
+  const sortByAlphabetically = () => {
+    setIsAlphabeticaly(false);
+    setIsLength(true);
+    setReverse(true);
+  };
+
+  const sortByLength = () => {
+    setIsAlphabeticaly(true);
+    setIsLength(false);
+    setReverse(true);
+  };
```

Comment:
if I first did the reverse and then sort it alphabetically or by length, then the reverse disappears. Please fix this